code/__HELPERS/atmos.dm

```diff
@@ -384,6 +383,8 @@ B --><-- A
 			target_drifting = FALSE
 			target_loc = target.loc
 
-		if((!drifting && user.loc != user_loc) || (!target_drifting && target.loc != target_loc) || (extra_checks && !extra_checks.Invoke()))
+		if((!(timed_action_flags & IGNORE_USER_LOC_CHANGE) && !drifting && user.loc != user_loc) \
+			|| (!(timed_action_flags& IGNORE_TARGET_LOC_CHANGE) && !target_drifting && target.loc != target_loc) \
+			|| (extra_checks && !extra_checks.Invoke()))
```

Comment:
this is a massive, impossible to read if statement

If it could be broken down into parts that are more readable maybe

Reply:
Figured it'd be readable enough with line breaks, but I can do that.